### PR TITLE
gpu: Fix protected memory on descriptor indexing

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/6937

Added 2 tests and found the protected memory check was wrapped in the wrong `if` so fixed that as well

### CODE REVIEW

File: tests/unit/gpu_av_descriptor_indexing.cpp [hunk 1]
```diff
@@ -2365,3 +2365,91 @@ TEST_F(NegativeGpuAVDescriptorIndexing, MultipleOOBTypesInOneCmdBuffer) {
     m_default_queue->Wait();
     m_errorMonitor->VerifyFound();
 }
+
+TEST_F(NegativeGpuAVDescriptorIndexing, MixingProtectedResources) {
+    TEST_DESCRIPTION("Have protected resources, but don't actually access it so no VU is triggered");
```
Comment (arno-lunarg) [inline tests/unit/gpu_av_descriptor_indexing.cpp:2370]:
```suggestion
    TEST_DESCRIPTION("Have protected resources, but don't actually access it so no VU is triggered");
```
Update description

**Review by arno-lunarg — APPROVED** [review]

File: layers/drawdispatch/descriptor_validator.cpp [hunk 2]
```diff
@@ -582,16 +582,17 @@ bool vvl::DescriptorValidator::ValidateDescriptor(const DescriptorBindingInfo &b
...
-            if (dev_state.ValidateProtectedImage(cb_state, *image_view_state->image_state, loc,
-                                                 vuids.unprotected_command_buffer_02707, " (Image is in a descriptorSet)")) {
-                return true;
-            }
-            if (variable->IsWrittenTo() &&
-                dev_state.ValidateUnprotectedImage(cb_state, *image_view_state->image_state, loc,
-                                                   vuids.protected_command_buffer_02712, " (Image is in a descriptorSet)")) {
-                return true;
-            }
+    }
```
Comment (jeremyg-lunarg) [inline layers/drawdispatch/descriptor_validator.cpp:585]: It took me a while to realize the only non-whitespace part of this change was moving the closing brace!

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM but the PR title is... odd

**Comment by spencer-lunarg** [comment]
@jeremyg-lunarg updated PR title

File: layers/drawdispatch/descriptor_validator.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply layers/drawdispatch/descriptor_validator.cpp:585]: ya, the negative test was a good thing to add, this logic was inside the parent `if ()` for no reason

**Pull request merged**